### PR TITLE
[Feature] Support HRNet and add pre-trained models

### PR DESCRIPTION
**Pull request by mzr1996**
## Motivation

Support HRNet

## Modification

As the title

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/_base_/schedules/imagenet_bs512_baseline.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Scheduler settings
```
Comment (Ezra-Yu) [inline configs/_base_/schedules/imagenet_bs512_baseline.py:1]: why call this file `xxx_baseline.py`?

Comment (mzr1996) [reply configs/_base_/schedules/imagenet_bs512_baseline.py]: They're files of other work, removed.

File: mmcls/models/backbones/hrnet.py [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+                    block=block,
+                    in_channels=self.in_channels[i],
+                    out_channels=out_channels,
+                    num_blocks=num_blocks[i],
+                ))
+
+        return ModuleList(branches)
+
+    def _make_fuse_layers(self):
+        if self.num_branches == 1:
```
Comment (Ezra-Yu) [inline mmcls/models/backbones/hrnet.py:80]: this function is  so long and complicated, please split it or add some comments.

**Review by Ezra-Yu — CHANGES_REQUESTED** [review]

File: mmcls/models/backbones/hrnet.py (hunk 2 above)
Comment (mzr1996) [reply mmcls/models/backbones/hrnet.py:105]: The logic of `make_fuse_layers` is a little complex. It's the same as the original code and hard to split. I added some comments to explain each part.

Comment (Ezra-Yu) [reply mmcls/models/backbones/hrnet.py:105]: ok.

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Pull request merged**